onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -273,7 +274,7 @@ Status QkvToContext(
     } else {
       // For fused TRT attention, transpose qkv to BxSxNx3xH. For fused causal kernel, transpose to 3xBxNxSxH.
       const int format = (use_fused_kernel ? 2 : 1);
-      T* qkv_add_bias = use_fused_causal ? data.gemm_buffer : nullptr;
+      qkv_add_bias = use_fused_causal ? data.gemm_buffer : nullptr;
       int matrix_to_transpose = ((!use_fused_kernel && past_present_share_buffer) ? 1 : 3);
```

Comment:
!use_fused_kernel => !use_fused_causal?

Reply:
We need to add gemm_buffer (all q k v) with bias when causal as they are all needed for the fused kernel.
Definetly there is no need to add kv bias when causal, yet this is the current logic. And this is bug fix only, not to improve performance here. We could improve it in future PR.